lib/libzfs/libzfs_sendrecv.c

```diff
@@ -2874,13 +2874,14 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 		zfs_handle_t *zhp;
 
 		/*
-		 * Destination fs exists.  Therefore this should either
+		 * Destination fs exists (and we have not been asked to
+		 * skip existing snapshots). Therefore this should either
 		 * be an incremental, or the stream specifies a new fs
 		 * (full stream or clone) and they want us to blow it
 		 * away (and have therefore specified -F and removed any
 		 * snapshots).
 		 */
-		if (stream_wantsnewfs) {
+		if (stream_wantsnewfs && !flags->skip) {
 			if (!flags->force) {
 				zcmd_free_nvlists(&zc);
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
```

Comment:
Let's update this error message to reference the optional `-s' option.  Something like "destination '%s' exists\nmust specify -F to overwrite it or -s to skip it"